src/device/device-management.ts

```diff
@@ -51,16 +54,30 @@ export type VideoDeviceConstraints = {
 export async function createCameraTrack(
   constraints?: VideoDeviceConstraints
 ): Promise<LocalCameraTrack> {
-  let stream: MediaStream;
+  // let stream: MediaStream;
+  if (isCameraTaskInProgress) {
+    throw new WcmeError(
+      ErrorTypes.CREATE_CAMERA_TRACK_FAILED,
+      `camera track can NOT be created since previous call has not been finished yet`
+    );
+  }
+  if (capturedVideoStream && capturedVideoStream.active) {
+    throw new WcmeError(
+      ErrorTypes.CREATE_CAMERA_TRACK_FAILED,
+      `camera track can NOT be created since previous captured video stream does not be stopped`
+    );
+  }
+  isCameraTaskInProgress = true;
```

Comment:
this solution won't work if the app calls media.getUserMedia directly

Reply:
Yeah, this solution can not solve this.